presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/framework/BenchmarkManager.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.benchmark.framework;
+
+import com.facebook.airlift.event.client.EventClient;
+import com.facebook.presto.benchmark.event.BenchmarkPhaseEvent;
+import com.facebook.presto.benchmark.event.BenchmarkSuiteEvent;
+import com.facebook.presto.benchmark.executor.PhaseExecutionException;
+import com.facebook.presto.benchmark.executor.PhaseExecutor;
+import com.facebook.presto.benchmark.executor.PhaseExecutorFactory;
+import com.facebook.presto.benchmark.source.BenchmarkSuiteSupplier;
+import com.google.inject.Inject;
+
+import javax.annotation.PostConstruct;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.benchmark.event.BenchmarkQueryEvent.EventStatus.FAILED;
+import static com.facebook.presto.benchmark.event.BenchmarkQueryEvent.EventStatus.SUCCEEDED;
+import static java.util.Objects.requireNonNull;
+
+public class BenchmarkManager
```

Comment:
Let's call it `BenchmarkRunner`.`BenchmarkManager` sounds like aa manger for benchmarks, not for running a benchmark.